ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java

```diff
@@ -442,31 +439,24 @@ protected Boolean findNextCompactionAndExecute(boolean collectGenericStats, bool
 
       // Don't start compaction or cleaning if not necessary
       if (isDynPartAbort(t, ci)) {
-        msc.markCompacted(CompactionInfo.compactionInfoToStruct(ci));
+        markWithState(ci, null, CompactionState.Compacted);
         compactionTxn.wasSuccessful();
         return false;
       }
       AcidDirectory dir = getAcidStateForWorker(ci, sd, tblValidWriteIds);
       if (!isEnoughToCompact(ci.isMajorCompaction(), dir, sd)) {
         if (needsCleaning(dir, sd)) {
-          msc.markCompacted(CompactionInfo.compactionInfoToStruct(ci));
+          markWithState(ci, null, CompactionState.Compacted);
         } else {
-          // do nothing
-          msc.markCleaned(CompactionInfo.compactionInfoToStruct(ci));
+          markWithState(ci, "None of the compaction thresholds hit, compaction request is refused!", CompactionState.Refused);
```

Comment:
I would use some diff state here like` ATTEMPTED`, meaning that it might be retried when criteria match. 
In the case of `REFUSED`, there is not much sense to retry as it identifies the case that is not supported.

Reply:
Since the compaction will be rescheduled when the table changes, REFUSED can be thought of 'The current state of the table is refused.' The main difference is whether we want to keep the original request and process it once possible, or just refuse it and process request only which fulfill all criteria at the time of checking.

Reply:
Well, ATTEMPTED is now called DID NOT INITIATE so I'm not sure it makes sense in this case. Also compaction isn't supported by hive in this case, and it doesn't make sense to retry (until more deltas are added).
But also I see your point that it's not an error/unsuccessful per se... (REFUSED compactions are represented in the "ratio of unsuccessful compactions" metric) ... so I'm not sure what ideal behavior should be.